Decent.Minecraft.Client/Java/JavaBlock.cs

```diff
@@ -80,7 +80,9 @@ static JavaBlock()
                 d == 1 ? new MossyStoneBricks() :
                 d == 2 ? new CrackedStoneBricks() :
                 (StoneBricks)new ChiseledStoneBricks();
-            _ctors[Id<Wood>()] = d => new Wood((WoodSpecies)(d & 0x3), (Axis)(d & 0xC));
+            // Wood is both 17 and 162 depending on species
+            _ctors[17] = d => new Wood((WoodSpecies)(d & 0x3), (Axis)(d & 0xC));
```

Comment:
Yeah, happy to move the magic number out into a constant. Sounds like you are in agreement that having a single wood type to represent all woods makes sense. The challenge with wood is that it has more than 4 types and overflowed into another block type. AcaciaWood isn't accurate because its also Dark Oak and will probably have another one added at some point in the future. Seems like a general pattern for things that have more than 4 types to be represented by different block ID's and the different ranges in the Enum values need to map to different block ID's.

Are you sure Snow is implemented correctly? Snow and Snow layer are different block types and have different characteristics. An 8 layer SnowLayer block isn't the same as a solid Snow Block. One Melts and the other doesn't.